### ISSUE TITLE
Players are suddenly unable to login (Snapshot Velocity version)

### ISSUE THREAD

**Issue by skerit** [AnvilPowered/Catalyst#55]
Suddenly, users are no longer able to login:

```
[11:39:23 INFO]: [connected player] Tester has disconnected: An internal error occurred in your connection.
[11:39:23 ERROR]: [connected player] Tester: exception encountered in com.velocitypowered.proxy.connection.client.ClientPlaySessionHandler@1391bfb
io.netty.handler.codec.EncoderException: java.lang.IllegalArgumentException: Don't know how to serialize org.anvilpowered.catalyst.common.util.CommonArgumentType
	at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:125) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:717) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:764) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:790) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:758) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1020) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:299) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at com.velocitypowered.proxy.connection.MinecraftConnection.write(MinecraftConnection.java:197) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at com.velocitypowered.proxy.connection.backend.BackendPlaySessionHandler.lambda$handle$2(BackendPlaySessionHandler.java:193) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.IllegalArgumentException: Don't know how to serialize org.anvilpowered.catalyst.common.util.CommonArgumentType
	at com.velocitypowered.proxy.protocol.packet.brigadier.ArgumentPropertyRegistry.serialize(ArgumentPropertyRegistry.java:89) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at com.velocitypowered.proxy.protocol.packet.AvailableCommands.serializeNode(AvailableCommands.java:148) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at com.velocitypowered.proxy.protocol.packet.AvailableCommands.encode(AvailableCommands.java:111) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at com.velocitypowered.proxy.protocol.netty.MinecraftEncoder.encode(MinecraftEncoder.java:34) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at com.velocitypowered.proxy.protocol.netty.MinecraftEncoder.encode(MinecraftEncoder.java:12) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:107) ~[1.1.x-SNAPSHOT.jar:1.1.3-SNAPSHOT (git-aef0e4a8-b329)]
	... 17 more

```

**Comment by alexstaeding** [comment]
I believe this issue has already been fixed in the dev branch by 72a4b42f310a5da3346c571ceaf214a372b095ab

You can build Catalyst from this branch or use the hotfix jar in our discord in the catalyst channel until catalyst v0.3.1 is released.

https://discord.gg/8RUzuwu

https://discord.com/channels/675484700357296138/675552711726596111/784084388576362516

**Comment by alexstaeding** [comment]
Catalyst [v0.3.1](https://github.com/AnvilPowered/Catalyst/releases/tag/v0.3.1) is out and has the fix for this.
